scripts/pcf_clang_format.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+"""
+    clang-format wrapper for pre-commit integration with external style-file
+
+    In clang-format-14 one can specify the path to clang-format-definitions,
+    however, clang-format-14 is at the time of writing this not released and
+    therefore a bit cumbersome to distribute. This script wraps the invocation
+    of clang-format in a manner such that it transforms the given file into
+    -style arguments as well as invoking clang-format in a manner such that it
+    reports issues, fixes them, and instructs the comitter how to see inspect
+    the auto-charges performed.
+
+    When running from shell, returns 0 on success, some other value otherwise
+"""
+import argparse
+import os
+import subprocess
+import sys
+
+
+def expand_path(path):
+    """Expands variables from the given path and turns it into absolute path"""
+
+    return os.path.abspath(os.path.expanduser(os.path.expandvars(path)))
+
+
+def parse_args():
+    """Parse command-line arguments"""
+
+    prsr = argparse.ArgumentParser(
+        description="Wrap clang-format",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    prsr.add_argument(
+        "--style-file",
+        help="Path to clang-format style definitions",
+        required=True,
+    )
+    prsr.add_argument("clang_format_args", nargs=argparse.REMAINDER)
+
+    args = prsr.parse_args()
+    args.style_file = expand_path(args.style_file)
+
+    return args
+
+
+def clang_format(args):
+    """Invoke clang-format"""
+
+    # Form arguments to clang-format parameter "-style"
+    style = []
+    with open(args.style_file) as sfd:
```

Comment:
Unpythonic variable naming: `sfd` -> `style_file`, `f` or like that

Reply:
Yeah, I see your point. My C abbreviation habits leaking in here. Fixing it.

Reply:
Fixed.